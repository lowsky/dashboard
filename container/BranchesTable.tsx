import React, { useContext } from 'react';

import UILibContext from '../components/UILibContext';
import { GithubRepo } from "../lib/types/resolvers";

export interface BranchesTableProps {
  repo?: GithubRepo
}
const BranchesTable: React.FC<BranchesTableProps> = ({ repo  }) => {
  const { BranchInfoRow } = useContext(UILibContext);

  return (
    <table className="table is-bordered is-striped is-hoverable">
      <thead>
                <tr>
                    <th>
                        <span className="fas fa-code-branch" />
                        <span>Branch</span>
                    </th>
                    <th>PR</th>
                    <th>Commit</th>
                </tr>
            </thead>
            <tbody>
                {(repo?.branches || []).map((branch, idx) => (
                  <BranchInfoRow key={idx} branch={branch} />
                ))}
            </tbody>
        </table>
    );
};

export default BranchesTable;
