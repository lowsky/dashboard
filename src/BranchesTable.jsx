import React from 'react';
import Relay from 'react-relay';

import BranchInfoRow from './BranchInfoRow.jsx';

export const BranchesTable = React.createClass({
    render: function() {
        const { repo } = this.props;
        return (
            <table className="table table-striped table-bordered table-condensed">
                <thead>
                <tr>
                    <th>Live application</th>
                    <th>Branch</th>
                    <th>Commit</th>
                </tr>
                </thead>
                <tbody>
                {
                    repo.branches.map(
                        branch => <BranchInfoRow branch={branch} />
                    )
                }
                </tbody>
            </table>
        );
    }
});

export default Relay.createContainer(BranchesTable, {
    fragments: {
        repo: () => Relay.QL`
            fragment on GithubRepo {
                branches {
                    ${BranchInfoRow.getFragment('branch')}
                }
            }
        `
    }
});
