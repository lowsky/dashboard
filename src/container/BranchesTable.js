import React from 'react';

import Ui from '../components';

const BranchesTable = props => {
    const { repo = { branches: [] } } = props;

    const BranchInfoRow = Ui.createBranchInfoRow(props);

    return (
        <table className="mdl-data-table mdl-js-data-table ">
            <thead>
                <tr>
                    <th className="mdl-data-table__cell--non-numeric">Branch</th>
                    <th className="mdl-data-table__cell--non-numeric">Commit</th>
                </tr>
            </thead>
            <tbody>
                {repo.branches.map((branch, idx) => <BranchInfoRow key={idx} branch={branch} />)}
            </tbody>
        </table>
    );
};

export default BranchesTable;
