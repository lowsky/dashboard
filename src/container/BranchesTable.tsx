import React, { useContext } from 'react';

import { UILibContext } from '../components';

const BranchesTable = ({repo = {branches:[]}}) => {const { BranchInfoRow } = useContext(UILibContext);

    return (
        <table className="table is-bordered is-striped is-hoverable">
            <thead>
                <tr>
                    <th className="is-narrow">
                        <span className="fas fa-code-branch" />
                        <span>Branch</span>
                    </th>
                    <th>Commit</th>
                </tr>
            </thead>
            <tbody>
                {(repo.branches || []).map((branch, idx) => (
                    <BranchInfoRow key={idx} branch={branch} />
                ))}
            </tbody>
        </table>
    );
};

export default BranchesTable;
