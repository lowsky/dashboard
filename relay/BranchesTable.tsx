import { createFragmentContainer } from 'react-relay';
import { graphql } from "react-relay";
import React from 'react';

import BranchesTable from '../container/BranchesTable';

export default createFragmentContainer(props => <BranchesTable {...props} />, {
    repo: graphql`
        fragment BranchesTable_repo on GithubRepo {
            branches {
                ...BranchInfoRow_branch
            }
        }
    `,
});
