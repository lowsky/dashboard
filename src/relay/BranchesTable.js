import { createFragmentContainer } from 'react-relay';
import graphql from 'babel-plugin-relay/macro';
import React from 'react';

import BranchesTable from '../container/BranchesTable';

export default createFragmentContainer(
    (props) => <BranchesTable {...props}/>,
    graphql`
        fragment BranchesTable_repo on GithubRepo {
            branches {
                ...BranchInfoRow_branch
            }
        }
    `
);
