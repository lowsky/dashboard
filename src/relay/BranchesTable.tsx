import { createFragmentContainer } from 'react-relay';
import graphql from 'babel-plugin-relay/macro';
import React from 'react';

import BranchesTable from '../container/BranchesTable';

// @ts-ignore
export default createFragmentContainer(props => <BranchesTable {...props} />, {
    repo: graphql`
        fragment BranchesTable_repo on GithubRepo {
            branches {
                ...BranchInfoRow_branch
            }
        }
    `,
});
