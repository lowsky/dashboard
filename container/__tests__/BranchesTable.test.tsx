import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import BranchesTable from '../BranchesTable';

describe('BranchesTable component', () => {
    it('should render without crashing', () => {
        const div = document.createElement('div');
        let repo = {
            branches: []
        };
        ReactDOM.render(<BranchesTable repo={repo} />, div);
    });

    it('should render as expected', () => {
        const props = { repo: {branches: [{}] }};
        const appContainer = renderer.create(<BranchesTable {...props} />).toJSON();
        expect(appContainer).toMatchSnapshot();
    });
});
