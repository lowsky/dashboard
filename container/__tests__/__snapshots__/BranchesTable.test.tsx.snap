// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BranchesTable component should render as expected 1`] = `
<table
  className="table is-bordered is-striped is-hoverable"
>
  <thead>
    <tr>
      <th>
        <span
          className="fas fa-code-branch"
        />
        <span>
          Branch
        </span>
      </th>
      <th>
        Commit
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
